TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/chassis/MecanumChassis.java

```diff
@@ -0,0 +1,51 @@
+package org.firstinspires.ftc.teamcode.subsystems.chassis;
+
+import com.arcrobotics.ftclib.command.SubsystemBase;
+import com.arcrobotics.ftclib.drivebase.MecanumDrive;
+import com.arcrobotics.ftclib.hardware.RevIMU;
+import com.arcrobotics.ftclib.hardware.motors.Motor;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+import org.firstinspires.ftc.teamcode.RobotHardwareMap;
+
+public class MecanumChassis extends SubsystemBase {
+    private static MecanumChassis instance;
+
+    private static MecanumDrive mecanumDrive;
+
+    private static RevIMU imu;
+
+    public static MecanumChassis getInstance() {
+        init();
+        return instance;
+    }
+
+    public static void init() {
+        if (instance == null) {
+            instance = new MecanumChassis();
+            initIMU();
+        }
+    }
+
+    private MecanumChassis() {
+        mecanumDrive = RobotHardwareMap.getInstance().mecanumDrive;
+        imu = RobotHardwareMap.getInstance().imu;
+    }
+
+    public void fieldCentricDrive(double strafeSpeed, double forwardSpeed, double theta) {
+        mecanumDrive.driveFieldCentric(strafeSpeed, forwardSpeed, theta, imu.getRotation2d().getDegrees());
+    }
+
+    public void robotCentricDrive(double strafeSpeed, double forwardSpeed, double theta) {
+        mecanumDrive.driveRobotCentric(strafeSpeed, forwardSpeed, theta);
+    }
+
+    public void resetHeading() {
+        imu.reset();
+    }
+
+    public static void initIMU() {
```

Comment:
this is not a static function, you can do init in the constructur or in init do instance.initImu()